### DISCUSSION TITLE
Trying to compare two text files and need help

### DISCUSSION THREAD

**Discussion by a84r7a3rga76fg** [WinMerge/winmerge#1365] [Q&A]
I've two text files and the important bit is the first 16 characters in every line. I want to write every line in the second text file to a third text file but only when the first 16 characters of the lines are identical in the first and second text files. Can someone please tell me how to do this?

First text file
`1123HF5DE3F15311 ?test*path\to1\file\test.exe`
`1123HF5DE3F1531A ?test*p2ath\to\file\test.exe`

Second text file
`1123HF5DE3F1531A ?test*drive\\te4st\random\file\test.exe`
`1123HF5DE3F14311 ?test*drive\\test\2random\file2\test.exe`

Third text file
`1123HF5DE3F1531A ?test*drive\\te4st\random\file\test.exe`

**Comment by sdottaka** [comment]
Ignore text differences from column 17 by setting and applying the IgnoreColumns prediffer plugin as shown below.
(If the First text file and Second text file are not sorted, also apply the SortAscending unpacker plugin.) 

![image](https://user-images.githubusercontent.com/98126/173207092-3eaa5d20-adb2-48b7-a96a-2cb3182419bf.png)
![image](https://user-images.githubusercontent.com/98126/173207082-4ae22750-040b-499a-90d4-bd1c4306696e.png)
![image](https://user-images.githubusercontent.com/98126/173207118-a09f952f-88b2-411b-bab9-c2825ead4ecd.png)

Use the Diff Context menu to show only the matching lines, select all the text on the right, and copy it into the Third text file.

![image](https://user-images.githubusercontent.com/98126/173207142-f2df85e8-dbaf-4331-b76e-34b5add4ddd9.png)

**Comment by a84r7a3rga76fg** [comment]
This didn't work for me, I also only have "Ignore Columns" instead of "Ignore Columns(IgnoreColumns)" in Prediffer under Plugins.

**Comment by sdottaka** [comment]
"Ignore Columns" selection is fine.
Please compare only the First text file and the Second text file, not the three files.
If that doesn't work, could you please attach a screenshot?

**Comment by a84r7a3rga76fg** [comment]
Had to censor it because the content is personal
![Untitled](https://user-images.githubusercontent.com/89746631/173209684-e7dc26a2-d5ec-48e0-bb0f-7048039220bc.png)
l

**Comment by sdottaka** [comment]
Lines with a light yellow background color indicate that the IgnoreColumns plugin ignores differences after the 17th character and that the first 16 characters identical.

If the first 16 characters are different but the background color is light yellow, please review the settings of the IgnoreColumns plugin.
